canonical-web-and-design/maas-ui
# Build read-only RepositoriesList component + scaffolding for remaining repos sections

## Done
- Built RepositoriesList component that shows all repos' names, urls, and enabled states (+ filtering and paginated at 20 per page)
- Built basic scaffolding for add and edit repository forms

## QA
- Go to `:8000/MAAS/settings/repositories`
- Check that the data in the table matches the data at `:5240/MAAS/#/settings/repositories`
- Check that the default repos are disabled (the first two)
- Check that filtering works as expected
- Change RepositoriesList prop `initialCount` to something low (e.g. 2) and check that pagination works as expected
- Click "Add PPA" or "Add repository" and check that the add form shows up (currently empty)
- Click the edit icon in a row and check that the edit form shows up (currently empty)
- Click the bin icon in a row and check that the row expands with delete confirmation

## Screenshots
### Default
![Screen Shot 2019-08-19 at 14 07 38-fullpage](https://user-images.githubusercontent.com/25733845/63264561-ac642e00-c28b-11e9-9350-eb01249337f3.png)


### Filter
![Screen Shot 2019-08-19 at 14 07 45-fullpage](https://user-images.githubusercontent.com/25733845/63264564-b0904b80-c28b-11e9-8528-469bcd677023.png)


### Delete
![Screen Shot 2019-08-19 at 14 07 52-fullpage](https://user-images.githubusercontent.com/25733845/63264572-b4bc6900-c28b-11e9-930f-1df05fc4f9a6.png)


### Add
![Screen Shot 2019-08-19 at 14 07 58-fullpage](https://user-images.githubusercontent.com/25733845/63264585-b9811d00-c28b-11e9-8fc0-e61221865d59.png)


### Edit
![Screen Shot 2019-08-19 at 14 08 02-fullpage](https://user-images.githubusercontent.com/25733845/63264587-bc7c0d80-c28b-11e9-9697-644907b6e590.png)
